test/libsolidity/util/TestFileParser.cpp

```diff
@@ -412,6 +414,31 @@ string TestFileParser::parseIdentifierOrTuple()
 	return identOrTuple;
 }
 
+void TestFileParser::validateType(std::string _type)
+{
+	auto validateBits = [&](string _baseType, int _bitsMin, int _bitsMax, bool _optional = false)
+	{
+		if (!boost::starts_with(_type, _baseType))
+			return;
+
+		if (_type.size() == _baseType.size())
+		{
+			if (_optional)
+				return;
+			else
+				throw Error(Error::Type::TypeError, "Missing bit width in '" + _type + "'");
+		}
+
+		int bits = std::stoi(_type.substr(_baseType.size()));
+		if (bits < _bitsMin || bits > _bitsMax)
+			throw Error(Error::Type::TypeError, "Invalid bit width in '" + _type + "'");
+	};
+
+	validateBits("int", 0, 256);
```

Comment:
Maybe change ``validateBits`` to ``[&](string, auto&& validator, bool _optional)`` and call it as
```suggestion
	validateBits("int", [](int _bits) { return 0 <= _bits && _bits <= 256 && (_bits % 8 == 0); });
	validateBits("uint", [](int _bits) { return 0 <= _bits && _bits <= 256 && (_bits % 8 == 0); });
	validateBits("bytes", [](int _bits) { return 0 <= _bits && _bits <= 32; }, true);
```
?

Reply:
Maybe call it ``validateIntSuffix`` then :-). Also why not make it ``unsigned``? Also ``std::from_chars`` is nicer than ``std::stoi``.

Reply:
well, it's not strictly speaking only int, as it also includes bool ;)
And maybe in the future it would be extended to fixed as well

Reply:
I just meant the suffix validator.

Reply:
I.e. the lambda.

Reply:
And ``IntSuffix`` as in somthing that has an integer suffix, not as in the type is an integer...

Reply:
it seems the CI doesn't know the `charconv` header (where from_chars is defined)

Reply:
:-(... it's standard C++17... but what can one do...